plugin/environment.go

```diff
@@ -291,6 +296,8 @@ func (env *Environment) Deactivate(id string) bool {
 		rp.supervisor.Shutdown()
 	}
 
+	env.registeredPlugins.Delete(id)
```

Comment:
Why do we delete the plugin from the map /and/ test `env.IsActive` everywhere else? Is anyone relying on the set of registered plugins to include inactive plugins? (i.e. do we use the environment in the system console when displaying what plugins are available?)

Reply:
It's possible to have a plugin that is registered but is _not_ active. An example would be when a plugin fails to start. Also deleting it makes for a cleaner re-registering the same plugin.

> Is anyone relying on the set of registered plugins to include inactive plugins? (i.e. do we use the environment in the system console when displaying what plugins are available?)

Yes, when displaying statuses that includes plugins that failed to start.

![image](https://user-images.githubusercontent.com/25732808/72938009-367f5980-3d38-11ea-8ca5-d4f119974a21.png)